algorithms/kernel/dtrees/forest/regression/oneapi/df_regression_tree_helper_impl.i

```diff
@@ -0,0 +1,160 @@
+/* file: df_regression_tree_helper_impl.i */
+/*******************************************************************************
+* Copyright 2014-2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+/*
+//++
+//  Implementation of the class defining the decision forest regression tree
+//--
+*/
+
+#ifndef __DF_REGRESSION_TREE_HELPER_IMPL__
+#define __DF_REGRESSION_TREE_HELPER_IMPL__
+
+#include "data_management/data/aos_numeric_table.h"
+#include "service/kernel/service_arrays.h"
+#include "algorithms/kernel/dtrees/dtrees_predict_dense_default_impl.i"
+
+namespace daal
+{
+namespace algorithms
+{
+namespace decision_forest
+{
+namespace regression
+{
+namespace internal
+{
+using namespace daal::algorithms::dtrees::internal;
+using namespace daal::services::internal;
+
+template <typename algorithmFPType, CpuType cpu>
```

Comment:
Do we have dependency from CPU here?

Reply:
yes, cpu is required in predict method
    algorithmFPType predict(const dtrees::internal::Tree & t, const algorithmFPType * x) const
    {
        const typename NodeType::Base * pNode = dtrees::prediction::internal::findNode<algorithmFPType, TreeType, cpu>(t, x);

Reply:
Be careful, I had the most pleasant and unexpected moments with this 😄